pygments/lexers/javascript.py

```diff
@@ -1534,3 +1536,63 @@ class JuttleLexer(RegexLexer):
         ]
 
     }
+
+
+class NodeConsoleLexer(Lexer):
+    """
+    For parsing JavaScript within an interactive Node.js shell, such as:
+
+    .. sourcecode:: nodejs
+
+        > let a = 3
+        undefined
+        > a
+        3
+        > let b = '4'
+        undefined
+        > b
+        '4'
+        > b == a
+        false
+    """
+    name = 'JavaScript Node.js console session'
+    aliases = ['nodejs', 'nodecon']
+    mimetypes = ['application/javascript', 'application/x-javascript',
+                 'text/x-javascript', 'text/javascript']
+
+    def get_tokens_unprocessed(self, text):
+        jslexer = JavascriptLexer(**self.options)
+
+        curcode = ''
+        insertions = []
+
+        for match in line_re.finditer(text):
+            line = match.group()
+            if line.startswith('> '):
+                insertions.append((len(curcode), 
+                    [(0, Generic.Prompt, line[:2])]))
+
+                curcode += line[2:]
+            elif line.startswith('...'):
+                # node does a nested ... thing depending on depth
+                code = line.lstrip('.')
```

Comment:
If you know it's `...`, wouldn't `line [4:]` be faster here, or is it possible that the prefix part is longer?

Reply:
> or is it possible that the prefix part is longer?

Yes
![image](https://user-images.githubusercontent.com/48163546/122672111-2b00be00-d1ca-11eb-804e-645c73b70550.png)